src/vs/workbench/contrib/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -596,8 +596,39 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 
 		// re-schedule this bit of the operation to be off the critical path - in case glob-match is slow
 		setImmediate(async () => {
+
+			let recommendationsToSuggest: string[] = [];
+			const now = Date.now();
+			forEach(this._availableRecommendations, entry => {
+				let { key: pattern, value: ids } = entry;
+				if (match(pattern, model.uri.path)) {
+					for (let id of ids) {
+						if (caseInsensitiveGet(product.extensionImportantTips, id)) {
+							recommendationsToSuggest.push(id);
+						}
+						const filedBasedRecommendation = this._fileBasedRecommendations[id.toLowerCase()] || { recommendedTime: now, sources: [] };
+						filedBasedRecommendation.recommendedTime = now;
+						if (!filedBasedRecommendation.sources.some(s => s instanceof URI && s.toString() === model.uri.toString())) {
+							filedBasedRecommendation.sources.push(model.uri);
+						}
+						this._fileBasedRecommendations[id.toLowerCase()] = filedBasedRecommendation;
+					}
+				}
+			});
+
+			this.storageService.store(
+				'extensionsAssistant/recommendations',
+				JSON.stringify(Object.keys(this._fileBasedRecommendations).reduce((result, key) => { result[key] = this._fileBasedRecommendations[key].recommendedTime; return result; }, {})),
+				StorageScope.GLOBAL
+			);
+
+			const config = this.configurationService.getValue<IExtensionsConfiguration>(ConfigurationKey);
+			if (config.ignoreRecommendations || config.showRecommendationsOnlyOnDemand) {
```

Comment:
It seems like this check should be done before setting recommendationsToSuggest to prevent doing the extra work.

Reply:
Hey Alex, I am not sure if that was intended or not. To give some history, I refactored this code a bit in 1.36. It was like this before that and hence I retained the same. Not sure what will break if I change the order :)

Reply:
Sounds good. I think for recovery less change is better, but it might be nice to move the check earlier in master.